docs/serving/services/creating-services.md

```diff
@@ -0,0 +1,63 @@
+---
+title: Creating Knative Services
+type: “docs”
+---
+
+Knative services are used to deploy an application. Knative Services attempt to model a more logical definition of a Service or Application. To achieve this, Services use the concept of Revisions which are each backed by a deployment with 2 Kubernetes Services.   
+
+For more information about Kubernetes Services, see the [Kubernetes documentation](https://kubernetes.io/docs/concepts/services-networking/service/).   
+
+Each Knative Service is defined by a Route and a Configuration, which have the same name as the Service, contained in a YAML file.
+
+## Before you begin
+
+To use Knative Services, you will need:
+* A Kubernetes cluster with [Knative installed](https://knative.dev/docs/install/index.html).
+* [Custom domains](https://knative.dev/docs/serving/using-a-custom-domain/) set up for Knative Services.
+
+## Creating a Knative Service
+To create an application, you need to create a YAML file that defines a Knative Service.
+This YAML file specifies metadata about the application, points to the hosted image of the app and allows the Service to be configured.
+
+This guide uses the [Hello World sample app in Go](https://knative.dev/docs/serving/samples/hello-world/helloworld-go) to demonstrate the structure of a Service YAML file and the basic workflow for deploying an app. These steps can be adapted for your own application if you have an image of it available on Docker Hub, Google Container Registry, or another container image registry.
+
+The Hello World sample app does the following:
+1. Reads an environment variable, `TARGET`, from the configuration .yaml file
+1. Prints `Hello World: \${TARGET}!`. If `TARGET` is not defined, it will print `NOT SPECIFIED`.
+
+### Procedure
+
+1. Create a new file named `service.yaml` containing the following information.
+  ```yml
+  apiVersion: serving.knative.dev/v1
+  kind: Service
+  metadata:
+   name: helloworld-go
+   namespace: default
+  spec:
+   template:
+    spec:
+     containers:
+      - image: gcr.io/knative-samples/helloworld-go
+       env:
+        - name: TARGET
+         value: "Go Sample v1"
+  ```
+  * `apiVersion`: The current Knative version.
+  * `name`(metadata): The name of the application.
+  * `namespace`: The namespace that the application will use.
+  * `image`: The URL to the image of the application.
+  * `name`(env): The environment variable printed out by the sample application.
+
+   > Note: If you’re deploying an image of your own app, update the name of the app and the URL of the image accordingly.
+
+## Modifying Knative Services
```

Comment:
This section doesn't show how to make an update to a Service, but it would be great to demo that.

Reply:
There are various example, if we want to demo how to update Label, we can write it like this:

---
For example, you can add a label to a Knative Service in the same way as Kubernetes resources.

```
$ kubectl label ksvc hello-example example=demo-label
```

Then, the label `example=demo-label` is copied to Route and Configuration automatically.

```
$ kubectl get ksvc,rt,config --show-labels
NAME                                        URL                                               LATESTCREATED           LATESTREADY             READY   REASON   LABELS
service.serving.knative.dev/hello-example   http://hello-example.default.3.1.165.238.xip.io   hello-example-hlplq-1   hello-example-hlplq-1   True             example=demo-label

NAME                                      URL                                               READY   REASON   LABELS
route.serving.knative.dev/hello-example   http://hello-example.default.3.1.165.238.xip.io   True             example=demo-label,serving.knative.dev/service=hello-example

NAME                                              LATESTCREATED           LATESTREADY             READY   REASON   LABELS
configuration.serving.knative.dev/hello-example   hello-example-hlplq-1   hello-example-hlplq-1   True             example=demo-label,serving.knative.dev/route=hello-example,serving.knative.dev/service=hello-example
```